src/github.com/travis-ci/jupiter-brain/vsphere.go

```diff
@@ -301,10 +302,27 @@ func (i *vSphereInstanceManager) createClient(ctx context.Context) (*govmomi.Cli
 	client.Client.RoundTripper = &soapBreakerRoundTripper{
 		rt: client.Client.RoundTripper,
 		cb: gobreaker.NewCircuitBreaker(gobreaker.Settings{
-			Name: "vSphere govmomi",
+			Name:     "vSphere govmomi",
+			Interval: 10 * time.Minute,
 			ReadyToTrip: func(counts gobreaker.Counts) bool {
 				failureRatio := float64(counts.TotalFailures) / float64(counts.Requests)
-				return counts.Requests >= 3 && failureRatio >= 0.6
+				return counts.Requests >= 5 && failureRatio >= 0.6
```

Comment:
Any time a request fails and the circuit breaker is closed, this function is called and if it returns true, the circuit breaker "trips" and goes into an open state. The "counts" are reset every 10 minutes, so the numbers include between 0 and 10 minutes of requests (which is the reason for the `counts.Requests >= 5` part, since otherwise the circuit breaker would trip immediately if the first request after a reset fails).